### ISSUE TITLE
Setting up without Docker

### ISSUE THREAD

**Issue by cfotache** [ApolloAuto/apollo#61]
I started installing it without Docker, on an Ubuntu VM where I have all my dev tools. I'm planning to use just simulator for now, not the full car hardware. Building the apollo-platform went well (had to install a few missing dependencies), then did the main apollo setup:

bash Apollo.sh build

I went through a few errors cause by missing dependencies, and now I ran into this:

```
ERROR: /home/cfotache/Python/Apollo/apollo/modules/prediction/BUILD:18:1: Linking of rule '//modules/prediction:prediction' failed: gcc failed: error executing command /usr/bin/gcc -o bazel-out/local-dbg/bin/modules/prediction/prediction '-Wl,-rpath,$ORIGIN/../../_solib_k8/_U@ros_S_S_Cros_Ucommon___Uexternal_Sros_Slib' ... (remaining 8 argument(s) skipped): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
modules/prediction/main.cc:23: error: undefined reference to 'ros::init(int&, char**, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int)'
external/ros/include/ros/publisher.h:107: error: undefined reference to 'ros::console::initializeLogLocation(ros::console::LogLocation*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, ros::console::levels::Level)'
external/ros/include/ros/publisher.h:107: error: undefined reference to 'ros::console::initializeLogLocation(ros::console::LogLocation*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, ros::console::levels::Level)'
external/ros/include/ros/publisher.h:107: error: undefined reference to 'ros::console::initializeLogLocation(ros::console::LogLocation*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, ros::console::levels::Level)'
external/ros/include/ros/publisher.h:113: error: undefined reference to 'ros::console::initializeLogLocation(ros::console::LogLocation*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, ros::console::levels::Level)'
modules/common/adapters/adapter_manager.cc:48: error: undefined reference to 'ros::NodeHandle::NodeHandle(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > const&)'
collect2: error: ld returned 1 exit status
```

Any idea how to get over this error?
Or any other specific recommendations, when not using Docker?

**Comment by startcode** [comment]
It looks like an environment problem. If you pull the latest code, and reproduce this problem,
you can run the script to get your environment 
```
bash scripts/env.sh >& env.txt
```
If you can share the contents in` env.txt`, We can investigate this problem further.
Thank you.

**Comment by cfotache** [comment]
Here it is:

```
cvltest
Outside docker
+ uname -a
Linux cvltest 4.4.0-83-generic #106-Ubuntu SMP Mon Jun 26 17:54:43 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
+ docker --version
scripts/env.sh: line 35: docker: command not found
+ docker images
+ grep apollo
scripts/env.sh: line 36: docker: command not found
+ echo -----------env---------------
-----------env---------------
+ env
XDG_SESSION_ID=24
SHELL=/bin/bash
TERM=xterm
SSH_CLIENT=199.255.244.65 40466 22
SSH_TTY=/dev/pts/0
USER=cfotache
LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
PATH=/home/cfotache/bin:/home/cfotache/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
MAIL=/var/mail/cfotache
PWD=/home/cfotache/Python/Apollo/apollo
LANG=en_US.UTF-8
HOME=/home/cfotache
SHLVL=2
LOGNAME=cfotache
SSH_CONNECTION=199.255.244.65 40466 10.0.0.6 22
APOLLO_HOME=/home/cfotache/Python/Apollo/apollo
LESSOPEN=| /usr/bin/lesspipe %s
XDG_RUNTIME_DIR=/run/user/1000
LESSCLOSE=/usr/bin/lesspipe %s %s
_=/usr/bin/env
```

**Comment by startcode** [comment]
First, make sure you have a valid `bazle-apollo/external/ros/` path. If not, it will be generated after you run
`bash apollo.sh build`, even you cannot build success at this time.

Then you source the `setup.bash` of ros.
```
source bazel-apollo/external/ros/setup.bash
```

Then try to build again by `bash apollo.sh build`.

Again, you are commended to compile inside docker. You still can read/edit the code outside docker.

Thanks

**Comment by cfotache** [comment]
I followed the instructions but it still didn't work, it stops with the same error.
Let me know if you have other suggestions, otherwise please leave the thread open in case we come with a solution soon, I'm sure there would be more people interested in this.

**Comment by startcode** [comment]
@cfotache you look at the operations in [docker/dev.dockerfile](https://github.com/ApolloAuto/apollo/blob/master/docker/dev.dockerfile)
and install all the packages on your host machine.

**Comment by xiaoxq** [comment]
@jamiehowarth
As you can imagine, the build and maintenance is really complicated and error prune. Due to our engineer resource bandwidth, we removed the dockerfile and just provide an out-of-the-box image.

**Comment by WindyCitySDR** [comment]
@xiaoxq, would you happen to have a download location of the "Image" for anyone to avoid the problems listed above ?

**Comment by xiaoxq** [comment]
Yeah, we have re-published the Dockerfile along with Apollo 2.5, which is located at /docker/build.
So you can build your own image and add / remove / fix any dependencies.
